typing/ctype.ml

```diff
@@ -3740,7 +3794,7 @@ and eqtype_kind k1 k2 =
   match k1, k2 with
   | (Fvar _, Fvar _)
   | (Fpresent, Fpresent) -> ()
-  | _                    -> raise_unexplained_for Equality
+  | _                    -> assert false
```

Comment:
What is the argument for this `assert false` ? Note if this `assert false` is justified,  the function `eqtype_kind` should be removed.

Reply:
You're right, this wasn't justified – I've gone back.  However, we've looked into this and we're fairly sure that OCaml code can't trigger this in practice (not just "doesn't", but "can't"), even though nothing in the structure of types prevents this; that said, we also can't demonstrate that.  So we've added a comment to that effect rather than a test case that raises the error.

Reply:
It does sound like it could be a potentially interesting simplification, but in a separate PR and after more investigations.